Formula/fluxctl.rb

```diff
@@ -0,0 +1,33 @@
+class Fluxctl < Formula
+  desc "fluxctl talks to Weave Flux and helps you deploy your code"
+  homepage "https://github.com/weaveworks/flux"
+  url "https://github.com/weaveworks/flux.git",
+      :tag => "1.7.1",
+      :revision => "185dfdf5a317d8ed2042f6afe2d6b94950282aff"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    arch = MacOS.prefer_64_bit? ? "amd64" : "x86"
+    dir = buildpath/"src/github.com/weaveworks/flux"
+    dir.install buildpath.children - [buildpath/".brew_home"]
+
+    cd dir do
+      system "dep", "ensure"
+      system "dep", "prune"
+
+      # Make binary
+      system "go", "build", "./cmd/fluxctl"
+      bin.install "fluxctl"
+
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/fluxctl 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
It will need a `flux` daemon to talk to. I'll find out from the dev team, if we can run some more tests.

Reply:
What do other API clients do with respect to testing? Set up a dummy server, maybe?

Reply:
This is what [kubectl does](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb#L50).

Reply:
Oh OK. We could run `fluxctl --version` to get the version; built like this, it'll say "unversioned", so perhaps we'll want to use the Makefile to build it (making adjustments to that, if necessary).

Reply:
@fxcoudert I updated the test and followed the lead of the `kubernetes-cli` formula. I tested it with the regular and the `--HEAD` install. Let me know if that works for you.